src/lib/dialog/dialog.md

```diff
@@ -126,3 +126,10 @@ that the AOT compiler knows to create the `ComponentFactory` for it.
 })
 export class AppModule() {}
 ```
+
+### Accessibility
+The dialog has `role="dialog"`, and dialog role can be overwritten to `alertdialog` in the
+appropriate context.
```

Comment:
```markdown
By default, each dialog has `role="dialog"` on the root element. The role can be changed to
`alertdialog` via the `MdDialogConfig` when opening.

The `aria-label`, `aria-labelledby`, and `aria-describedby` attributes can all be set to the
dialog element via the `MdDialogConfig` as well. Each dialog should typically have a label
set via `aria-label` or `aria-labelledby`.

#### Focus management
By default, the first tabbable element within the dialog will receive focus upon open. 

Tabbing through the elements of the dialog will keep focus inside of the dialog element, 
wrapping back to the first tabbable element when reaching the end of the tab sequence.
```
(ariaLabel being added in #6558)

We can expand this in a later PR to mention the focus control attributes from `cdkFocusTrap` once we have documentation for that